compiler_opt/rl/feature_ops_test.py

```diff
@@ -104,6 +143,65 @@ def test_normalize_fn_sqrt_z_normalization(self, with_sqrt, with_z_score,
     self.assertAllEqual(expected_shape, output.shape)
     self.assertAllClose(expected, output)
 
+  def test_get_ir2vec_normalize_fn(self):
+    normalize_fn = feature_ops.get_ir2vec_normalize_fn()
+    obs = tf.constant(value=[[2.0], [8.0]])
+    output = normalize_fn(obs)
+    self.assertAllEqual(obs.shape, output.shape)
+    self.assertAllClose(obs, output)
+
+    normalize_fn = feature_ops.get_ir2vec_normalize_fn(
+        ir2vec_with_z_score_normalization=True)
+    output = normalize_fn(obs)
+    expected = np.array([[-1], [1]])
+    self.assertAllEqual(obs.shape, output.shape)
+    self.assertAllClose(expected, output)
+
+  def test_get_ir2vec_dimensions_from_vocab_file_valid_file(self):
+    vocab_data = {
+        'section1': {
+            'instruction1': [1.0, 2.0, 3.0, 4.0, 5.0],  # 5 dimensions
+            'instruction2': [2.0, 3.0, 4.0, 5.0, 6.0]
+        },
+        'section2': {
+            'instruction3': [0.1, 0.2, 0.3, 0.4, 0.5]
+        }
+    }
+    self._test_vocab_file_dimensions(vocab_data, 5)
+
+  @parameterized.named_parameters(
+      ('empty_embedding', {
+          'section1': {
+              'instruction1': []
+          }
+      }, ValueError),
+      ('invalid_json', 'invalid json content {', json.JSONDecodeError),
```

Comment:
Here and below: I'm not sure testing that `json.parse` throws a decode error on invalid JSON is useful here. Same for the file not existing.